File: apps/cargo-tracker/src/shared/infrastructure/auth/bcrypt-password-verifier.ts
import bcrypt from 'bcryptjs';
import type { PasswordVerifier } from './user-account.js';

/** bcrypt によるパスワード照合・ハッシュ生成アダプター */
export class BcryptPasswordVerifier implements PasswordVerifier {
  async verify(rawPassword: string, passwordHash: string): Promise<boolean> {
    return bcrypt.compare(rawPassword, passwordHash);
  }

  /** シード・登録用のハッシュ生成 */
  static async hash(rawPassword: string): Promise<string> {
    return bcrypt.hash(rawPassword, 10);
  }
}

File: docs/article/source/java-3/apps/backend/authms/src/main/java/com/example/authms/infrastructure/security/BCryptPasswordVerifier.java
package com.example.authms.infrastructure.security;

import com.example.authms.application.port.PasswordVerifier;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Component;

@Component
public class BCryptPasswordVerifier implements PasswordVerifier {

    private final BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();

    @Override
    public boolean matches(String rawPassword, String passwordHash) {
        return encoder.matches(rawPassword, passwordHash);
    }
}
